main.go

```diff
@@ -202,13 +202,18 @@ func main() {
 	prometheus.MustRegister(collector.NewNodes(logger, httpClient, esURL, *esAllNodes, *esNode))
 
 	if *esExportIndices || *esExportShards {
-		prometheus.MustRegister(collector.NewShards(logger, httpClient, esURL))
+		sC := collector.NewShards(logger, httpClient, esURL)
+		prometheus.MustRegister(sC)
 		iC := collector.NewIndices(logger, httpClient, esURL, *esExportShards, *esExportIndexAliases)
 		prometheus.MustRegister(iC)
-		if registerErr := clusterInfoRetriever.RegisterConsumer(iC); registerErr != nil {
+		if clusterInfoRetriever.RegisterConsumer(iC) != nil {
 			level.Error(logger).Log("msg", "failed to register indices collector in cluster info")
 			os.Exit(1)
 		}
+		if clusterInfoRetriever.RegisterConsumer(sC) != nil {
```

Comment:
again, this isn't idiomatic
```suggestion
		if registerErr := clusterInfoRetriever.RegisterConsumer(sC); registerErr != nil {
```